### DISCUSSION TITLE
How to make Tailwind CSS utilities be loaded after CSS Modules and override them?

### DISCUSSION THREAD

**Discussion by yandodov** [vercel/next.js#59286] [Help]
### Summary

I have successfully set up CSS Modules and Tailwind CSS. Both work perfectly on their own. I can even use `@apply` in my modules files:

```css
.my-module {
	@apply bg-red;
}
```

However, the following doesn't work:

```jsx
import styles from "./my-module.module.css";

export default function MyModule() {
	return <div className={`${styles["my-module"]} bg-blue`}></div>;
}
```

This happens because the styles of CSS modules are loaded after the Tailwind styles. Ideally, the CSS modules output should be put as `@layer components`. I tried doing it like that:

```css
@layer components {
	.my-module {
		@apply bg-red;
	}
}
```

…but it didn't work:

> **Syntax error**: /Users/hristiyan.dodov/Projects/my-project/components/my-module.module.css `@layer components` is used but no matching `@tailwind components` directive is present.

This makes sense, because Next's CSS modules bundle is a completely separate thing from my global stylesheet containing the Tailwind directives.

Is there a way to fix this?

---

Perhaps in the ideal scenario, Next would generate 3 output files:

1. **layout.css** containing Tailwind CSS `base` and `components` layers
2. **page.css** containing the CSS modules, as usual
3. **utilities.css** containing the Tailwind CSS `utilities` layer

This would allow Tailwind utilities to override CSS modules. In the current setup, that's not possible:

1. **layout.css** contains all Tailwind layers, including `utilities`
2. **page.css** contains CSS modules, which override the utilities

**Comment by sina-salahshour** [comment]
i think this is more of a tailwind/postcss issue. i'm facing the same problem in a sveltekit project
